src/components/charts/modal/CreateChartGroup.tsx

```diff
@@ -27,19 +27,50 @@ export default class CreateChartGroup extends Component<CreateChartGroupProps, C
     }
 
     handleNameChange(event) {
-        this.setState({ name: {value: event.target.value, error: ""} });
+        this.setState({ name: { value: event.target.value, error: [] } });
     }
 
     handleDescriptionChange(event) {
         this.setState({ description: event.target.value });
     }
 
     async saveChartGroup(e) {
-        const nameRegexp = new RegExp(`^[a-z]+[a-z0-9\-\?]*[a-z0-9]+$`)
-        if (!nameRegexp.test(this.state.name.value)) {
-            this.setState({ name: { ...this.state.name, error: 'name must follow `^[a-z]+[a-z0-9\-\?]*[a-z0-9]+$` pattern.'}})
-            return
+        const lowercaseRegexp = new RegExp('^[a-z0-9-. ][a-z0-9-. ]*[a-zA-Z0-9-. ]$')
+        const startAndEndAlphanumericRegex = new RegExp(`^[a-zA-Z0-9].*[a-z0-9A-Z]$`)
+        const spaceNotAllowedRegex = new RegExp('^[a-zA-Z0-9!@#\$%\^\&*\)\(+=._-]+$')
```

Comment:
instead of this.. just search indexOf(" ")  less error prone and faster

Reply:
okay